### ISSUE TITLE
BBOT Freezes

### ISSUE THREAD

**Issue by Sh4d0wHunt3rX** [blacklanternsecurity/bbot#1244]
Today, twice my scan froze on tesla and openai. This is the last lines of debug.log

```
downloaded to /root/.bbot/scans/openai/2024-04-09_17-27-48/filedownload/20240409_1747_02_https-privacy-openai-com-main-penumbra.js
2024-04-09 17:47:07,725 [INFO] bbot.modules.filedownload base.py:1262 Found "footnote-9e6a21e04868e13dc75adf73cfaab1be722d32b46af9a8b1894712c647ab6499.br.js" at "https://global.discourse-cdn.com", downloaded to /root/.bbot/scans/openai/2024-04-09_17-27-48/filedownload/20240409_1747_04_https-global-discourse-cdn-com-openai1-assets-plugins-footnote-9e6a21e04868e13dc75adf73cfaab1be722d32b46af9a8b1894712c647ab6499-br.js
2024-04-09 17:47:09,817 [TRACE] bbot.core.helpers.web web.py:689 Unhandled exception with request to URL: https://github.com/date-fns/date-fns/blob/master/docs/upgradeGuide.md: Client error '404 Not Found' for url 'https://github.com/date-fns/date-fns/blob/main/docs/upgradeGuide.md'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/404
```

BBot v1.1.7.3218rc

Nothing more prints

![image](https://github.com/blacklanternsecurity/bbot/assets/15929497/0912709a-846e-48a9-9b3d-bd8e6bc067a8)

**Comment by Sh4d0wHunt3rX** [comment]
Another target today, nothing specific in debug.log. Nothing prints more on terminal by bbot

![image](https://github.com/blacklanternsecurity/bbot/assets/15929497/769d102f-04fc-4c64-8598-112726376419)

I also got this when stopped it by ctrl + c and it didn't finish properly, so I killed tmux pane.

![image](https://github.com/blacklanternsecurity/bbot/assets/15929497/69719086-462f-4cac-b120-5fa01073e849)

**Comment by TheTechromancer** [comment]
Can you post the entire debug.log?

**Comment by TheTechromancer** [comment]
Also is this a duplicate of https://github.com/blacklanternsecurity/bbot/issues/1053? Or a new issue?

**Comment by Sh4d0wHunt3rX** [comment]
[Copy of debug.log](https://github.com/blacklanternsecurity/bbot/files/14933959/Copy.of.debug.log)

Thanks, this is the debug of openai scan.

This issue is different than #1053 , that one was crashing, but this one just stops and won't print anything else.

**Comment by TheTechromancer** [comment]
I think we've been able to reproduce this. In our case, BBOT was frozen and one CPU core was stuck at 100%. It was completely frozen; we weren't able to cancel it via CTRL+C (which sadly means we weren't able to see what it was doing when it froze).

Based on this I have a couple of guesses as to what this might be:
- A race condition in one of our libraries (in the past we've had similar issues in dnspython and httpx)
- A regex denial-of-service (it's possible there's a certain webpage that's reacting with one of our regexes)

Sadly until we have a reliable way to reproduce this (preferably with a single target + single bbot module), there's not a lot we can do to fix it.

**Comment by Sh4d0wHunt3rX** [comment]
Thanks a lot for checking this. Only happened twice for me. 🙏

**Comment by TheTechromancer** [comment]
Finally tracked down this bug:
https://github.com/blacklanternsecurity/bbot/issues/1414.

We pushed a fix to dev and stable.

**Comment by Sh4d0wHunt3rX** [comment]
BBOT frozen for 15 minutes for now, not sure if it will continue or not. Scan duration was around 4 hours.

![image](https://github.com/blacklanternsecurity/bbot/assets/15929497/663a2acb-e572-430a-8b50-f408727ad631)
![image](https://github.com/blacklanternsecurity/bbot/assets/15929497/89ff15f8-8f0e-4598-a5db-338f4904b8e4)

[debug.log](https://github.com/user-attachments/files/15975766/debug.log)

**Comment by Sh4d0wHunt3rX** [comment]
Activity detected 20 mins later

![image](https://github.com/blacklanternsecurity/bbot/assets/15929497/86b3ada7-55bd-41df-b1f0-d14d0db8e1d3)

[debug.log](https://github.com/user-attachments/files/15975868/debug.log)

Keeps freezing and resuming

![image](https://github.com/blacklanternsecurity/bbot/assets/15929497/38705945-bb87-498d-9f36-544274998e3e)

**Comment by TheTechromancer** [comment]
Thanks @amiremami. This is most likely an issue with excavate, and will be fixed in bbot-2.0 as soon as https://github.com/blacklanternsecurity/bbot/pull/1471 is merged.

There was a bug with excavate where sometimes its regexes would take a long time.
